src/psyclone/psyir/backend/fortran.py

```diff
@@ -720,7 +822,8 @@ def gen_decls(self, symbol_table, args_allowed=True):
             # RoutineSymbols of UnknownFortranType. These must therefore be
             # declared.
             if isinstance(sym.datatype, UnknownType):
-                declarations += self.gen_vardecl(sym)
+                declarations += self.gen_vardecl(
+                    sym, include_visibility=(not args_allowed))
```

Comment:
It took me a while to figure out what was the relation between args_allowed and including visibility attributes, but I think it is because Containers don't allow arguments and routines don't allow visibility attributes. So it is just a way to see if we are in a routine or in a container scope. Is that the relation?

If that is the case, I suggest rename the args_allowed argument to is_routine_scope , or better yet switch it to is_module_scope=False as currently all uses have to use the negative of it.

Reply:
Sorry about that and yes, that is the relation. I've followed your suggestion and renamed it to `is_module_scope`.